keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -44,82 +45,83 @@ class ResourceActuator(
   private val log by lazy { LoggerFactory.getLogger(javaClass) }
 
   suspend fun <T : ResourceSpec> checkResource(resource: Resource<T>) {
-    val id = resource.id
-    val plugin = handlers.supporting(resource.apiVersion, resource.kind)
-
-    // todo eb: extract "application veto" as its own concept, and check before generating the diff.
-
-    if (plugin.actuationInProgress(resource)) {
-      log.debug("Actuation for resource {} is already running, skipping checks", id)
-      publisher.publishEvent(ResourceCheckSkipped(resource.apiVersion, resource.kind, id, "ActuationInProgress"))
-      return
-    }
-
-    try {
-      val (desired, current) = plugin.resolve(resource)
-      val diff = ResourceDiff(desired, current)
-      if (diff.hasChanges()) {
-        diffFingerprintRepository.store(id, diff)
+    withResourceTracingContext(resource) {
```

Comment:
I'm concerned that `MDC.remove(X_SPINNAKER_RESOURCE_ID)` doesn't appear in this PR. Other beans may run on a thread that formerly ran `ResourceActuator`, so it's good practice to clean up thread locals. A pattern that has become common in other services is to add to MDC via a try/finally block, with cleanup in the finally.

Reply:
That's good feedback, @asher, thanks! Originally I was playing around with just adding `MDCContex()` to coroutine builders like `async` and `launch` in specific places, which had the behavior of propagating current MDC into the coroutine and restoring the original context outside that scope, but it seems like with `withContext` that's not the case. Let me take yet another look at the library to make sure I got this right (maybe even add some tests).

Reply:
@asher you were spot on, thanks! I added a `try/catch` block to remove the tracing field as suggested. Please review.